crates/arroyo-api/src/lib.rs

```diff
@@ -142,12 +145,28 @@ pub async fn start_server(database: DatabaseSource, guard: ShutdownGuard) -> any
         ),
     );
 
-    info!("Starting API server on {:?}", local_addr);
-    guard.into_spawn_task(wrap_start("api", local_addr, async {
-        axum::serve(listener, app.into_make_service()).await
-    }));
+    if let Some(tls_config) = config.get_tls_config(&config.api.tls) {
```

Comment:
This pattern is repeated in all the services that startup TLS.  Maybe it should be part of `wrap_start`, with optional TLS configuration passed in?